### DISCUSSION TITLE
vercel-azure-devops-extension || Improvement Request: Enhancing Error Message for Expired Vercel Tokens

### DISCUSSION THREAD

**Discussion by SomnathKadam** [vercel/community#3256] [Idea]
Dear Team,

I wanted to bring to your attention that I encountered a misleading error message when my Vercel token expired. I believe there is room for improvement in the error messaging system. Upon the expiration of my token, I received the following error message:

`Error: Could not retrieve Project Settings. To link your Project, remove the `.vercel` directory and deploy again.`

To ensure a better user experience and clarity, I kindly request that you review and enhance the error message for situations when a Vercel token expires.

**Comment by amyegan** [comment]
Thanks for sharing this! I'll pass it along to the team 🙂

**Comment by cb1kenobi** [comment]
This isn't a ADO specific error. I've seen it happen on my local machine. I see the code does handle the scenarios when the token is missing or the token is "invalid". I don't know if "expired" means "invalid". We'll need to investigate.

**Reply by SomnathKadam** [reply]
The error message "Token expired" indicates that the token's date has passed, specifically referring to the expiration date. To illustrate this further, consider the following scenario:

Suppose today's date is July 25, 2025, and in the provided screenshot, the token's "Expires" date is set to July 24, 2025. In this case, the error mentioned above will be triggered.

![image](https://github.com/vercel/community/assets/10321563/d6fcb847-0af8-4da6-92a1-097788a4fb3d)
